nodes/spatial/voronoi_on_solid_mk2.py

```diff
@@ -0,0 +1,203 @@
+
+    def sv_init(self, context):
+        self.inputs.new('SvSolidSocket', 'Solid')
+        self.inputs.new('SvVerticesSocket', "Sites")
+        self.inputs.new('SvStringsSocket', "Inset").prop_name = 'inset'
+        self.outputs.new('SvSolidSocket', "InnerSolid")
+        self.outputs.new('SvSolidSocket', "OuterSolid")
+
+    def draw_buttons(self, context, layout):
+        layout.prop(self, "mode")
+        layout.prop(self, "flat_output")
+
+    def draw_buttons_ext(self, context, layout):
+        self.draw_buttons(context, layout)
+        layout.prop(self, 'scale_center')
+        layout.prop(self, 'accuracy')
+
+    def scale_cells(self, verts, sites, insets, precision):
+        if all(i == 0.0 for i in insets):
+            return verts
+        verts_out = []
+        for vs, site, inset in zip(verts, sites, insets):
+            if inset >= 1.0:
+                continue
+            if self.scale_center == 'SITE':
+                c = site
+            else:
+                c = center(vs)
+            vs1 = scale_relative(vs, c, 1.0 - inset)
+            if diameter(vs1, axis=None) <= precision:
+                continue
+            verts_out.append(vs1)
+        return verts_out
+
+    def process(self):
+
+        if not any(socket.is_linked for socket in self.outputs):
+            return
+
+        solid_in = self.inputs['Solid'].sv_get()
+        sites_in = self.inputs['Sites'].sv_get()
+        inset_in = self.inputs['Inset'].sv_get()
+
+        solid_in = ensure_nesting_level(solid_in, 2, data_types=(Part.Shape,))
+        input_level = get_data_nesting_level(sites_in)
+        sites_in = ensure_nesting_level(sites_in, 4)
+        inset_in = ensure_min_nesting(inset_in, 2)
+
+        nested_output = input_level > 3
+        need_inner = self.outputs['InnerSolid'].is_linked
+        need_outer = self.outputs['OuterSolid'].is_linked
+
+        precision = 10 ** (-self.accuracy)
+
+        inner_fragments_out = []
+        outer_fragments_out = []
+        for params in zip_long_repeat(solid_in, sites_in, inset_in):
+            new_inner_fragments = []
+            new_outer_fragments = []
+            for solid, sites, inset in zip_long_repeat(*params):
+                #verts, edges, faces = voronoi_on_solid(solid, sites, do_clip=True, clipping=None)
+                # see more info: https://github.com/nortikin/sverchok/pull/4977
+                box = solid.BoundBox
+                clipping = 1
+                x_min, x_max = box.XMin - clipping, box.XMax + clipping
+                y_min, y_max = box.YMin - clipping, box.YMax + clipping
+                z_min, z_max = box.ZMin - clipping, box.ZMax + clipping
+                bounds = list(itertools.product([x_min,x_max], [y_min, y_max], [z_min, z_max]))
+                verts, edges, faces = voronoi_on_mesh_bmesh(bounds, [ [0,1,3,2], [2,3,7,6], [6,7,5,4], [4,5,1,0], [2,6,4,0], [7,3,1,5] ], len(sites), sites, spacing=inset, mode='VOLUME' )
```

Comment:
numbers in brackets look magic :)
I'd suggest to assign them to a separate variable and add a comment about what is this.

Reply:
What brackets? If an argument for voronoi_on_mesh_bmesh then it is a cube faces that is a bounding box.